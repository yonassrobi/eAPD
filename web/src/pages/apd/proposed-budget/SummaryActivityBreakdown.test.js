import { shallow } from 'enzyme';
import React from 'react';
import {
  plain as SummaryActivityBreakdownTable,
  mapStateToProps
} from './SummaryActivityBreakdown';

const props = {
  ffy: '2016',
  activityIndex: 2,
  activityName: 'activity name',
  costSummary: {
    total: {
      federalShare: 100,
      medicaidShare: 200,
      otherFunding: 300,
      stateShare: 400,
      totalCost: 500
    },
    years: {
      2016: {
        contractorResources: [
          {
            description: 'contractor 2.1',
            totalCost: 2000,
            unitCost: null,
            units: null
          },
          {
            description: 'contractor 2.2',
            totalCost: 2001,
            unitCost: 20,
            units: '20 hours'
          }
        ],
        contractorResourcesTotal: 1000,
        federalPercent: 1001,
        federalShare: 90,
        keyPersonnel: [
          {
            description: 'key person (APD Key Personnel)',
            totalCost: 1002,
            unitCost: null,
            units: '100% time'
          }
        ],
        medicaidShare: 1003,
        nonPersonnel: [
          {
            description: 'costly costs',
            totalCost: 3000,
            unitCost: null,
            units: null
          }
        ],
        nonPersonnelTotal: 1004,
        otherFunding: 1005,
        statePercent: 10,
        statePersonnel: [
          {
            description: 'personnel role',
            totalCost: 4000,
            unitCost: 40,
            units: '0.5 FTE'
          }
        ],
        statePersonnelTotal: 1006,
        stateShare: 1007,
        totalCost: 1008
      },
      2017: {
        contractorResources: [
          {
            description: 'contractor 2.1',
            totalCost: 6000,
            unitCost: null,
            units: null
          },
          {
            description: 'contractor 2.2',
            totalCost: 6001,
            unitCost: 60,
            units: '60 hours'
          }
        ],
        contractorResourcesTotal: 7000,
        federalPercent: 7001,
        federalShare: 50,
        keyPersonnel: [
          {
            description: 'key person (APD Key Personnel)',
            totalCost: 8002,
            unitCost: null,
            units: '100% time'
          }
        ],
        medicaidShare: 7003,
        nonPersonnel: [
          {
            description: 'costly costs',
            totalCost: 9000,
            unitCost: null,
            units: null
          }
        ],
        nonPersonnelTotal: 7004,
        otherFunding: 7005,
        statePercent: 50,
        statePersonnel: [
          {
            description: 'personnel role',
            totalCost: 9000,
            unitCost: 90,
            units: '0.5 FTE'
          }
        ],
        statePersonnelTotal: 7006,
        stateShare: 7007,
        totalCost: 7008
      }
    }
  },
  otherFunding: {
    2016: {
      statePersonnel: 100,
      expenses: 200,
      contractors: 350,
      total: 650
    },
    2017: {
      statePersonnel: 400,
      expenses: 700,
      contractors: 150,
      total: 1250
    }
  },
  fundingSource: 'HIE'
};

describe('State and Contractor Cost Breakdown Table renders correctly', () => {
  it('renders correctly', () => {
    const component = shallow(<SummaryActivityBreakdownTable {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('maps redux state to component props', () => {
    const getActivity = jest.fn();
    getActivity.mockReturnValue({ key: 'activity key', name: 'activity name' });

    const getCostSummary = jest.fn();
    getCostSummary.mockReturnValue('cost summary');

    expect(
      mapStateToProps(
        'my state object',
        { activityIndex: 0 },
        { getActivity, getCostSummary }
      )
    ).toEqual({
      activityName: 'activity name',
      costSummary: 'cost summary'
    });

    expect(getActivity).toHaveBeenCalledWith('my state object', {
      activityIndex: 0
    });
    expect(getCostSummary).toHaveBeenCalledWith('my state object', {
      activityIndex: 0
    });

    // Now test that it builds a default activity name if none is provided
    getActivity.mockReturnValue({ key: 'activity key', name: '' });

    expect(
      mapStateToProps(
        'my state object',
        { activityIndex: 0 },
        { getActivity, getCostSummary }
      )
    ).toEqual({
      activityName: '',
      costSummary: 'cost summary'
    });
  });
});
