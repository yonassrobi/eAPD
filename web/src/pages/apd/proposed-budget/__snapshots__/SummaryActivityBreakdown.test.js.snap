// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`State and Contractor Cost Breakdown Table renders correctly renders correctly 1`] = `
<table
  className="budget-table activity-budget-table"
  id="activity-3-2016"
>
  <thead>
    <tr
      className="budget-table--row__primary-header"
    >
      <th
        scope="col"
      >
        Activity 
        3
         
        <span
          style={
            Object {
              "fontWeight": "100",
            }
          }
        >
          activity name
           (HIE)
        </span>
      </th>
      <th
        colSpan="4"
        scope="col"
      >
        Personnel Cost Ã— FTE
      </th>
      <th
        className="ds-u-text-align--right"
        scope="col"
      >
        Total cost
      </th>
    </tr>
  </thead>
  <tbody>
    <CostAllocationRows
      activityIndex={2}
      ffy="2016"
      otherFunding={
        Object {
          "2016": Object {
            "contractors": 350,
            "expenses": 200,
            "statePersonnel": 100,
            "total": 650,
          },
          "2017": Object {
            "contractors": 150,
            "expenses": 700,
            "statePersonnel": 400,
            "total": 1250,
          },
        }
      }
      years={
        Object {
          "2016": Object {
            "contractorResources": Array [
              Object {
                "description": "contractor 2.1",
                "totalCost": 2000,
                "unitCost": null,
                "units": null,
              },
              Object {
                "description": "contractor 2.2",
                "totalCost": 2001,
                "unitCost": 20,
                "units": "20 hours",
              },
            ],
            "contractorResourcesTotal": 1000,
            "federalPercent": 1001,
            "federalShare": 90,
            "keyPersonnel": Array [
              Object {
                "description": "key person (APD Key Personnel)",
                "totalCost": 1002,
                "unitCost": null,
                "units": "100% time",
              },
            ],
            "medicaidShare": 1003,
            "nonPersonnel": Array [
              Object {
                "description": "costly costs",
                "totalCost": 3000,
                "unitCost": null,
                "units": null,
              },
            ],
            "nonPersonnelTotal": 1004,
            "otherFunding": 1005,
            "statePercent": 10,
            "statePersonnel": Array [
              Object {
                "description": "personnel role",
                "totalCost": 4000,
                "unitCost": 40,
                "units": "0.5 FTE",
              },
            ],
            "statePersonnelTotal": 1006,
            "stateShare": 1007,
            "totalCost": 1008,
          },
          "2017": Object {
            "contractorResources": Array [
              Object {
                "description": "contractor 2.1",
                "totalCost": 6000,
                "unitCost": null,
                "units": null,
              },
              Object {
                "description": "contractor 2.2",
                "totalCost": 6001,
                "unitCost": 60,
                "units": "60 hours",
              },
            ],
            "contractorResourcesTotal": 7000,
            "federalPercent": 7001,
            "federalShare": 50,
            "keyPersonnel": Array [
              Object {
                "description": "key person (APD Key Personnel)",
                "totalCost": 8002,
                "unitCost": null,
                "units": "100% time",
              },
            ],
            "medicaidShare": 7003,
            "nonPersonnel": Array [
              Object {
                "description": "costly costs",
                "totalCost": 9000,
                "unitCost": null,
                "units": null,
              },
            ],
            "nonPersonnelTotal": 7004,
            "otherFunding": 7005,
            "statePercent": 50,
            "statePersonnel": Array [
              Object {
                "description": "personnel role",
                "totalCost": 9000,
                "unitCost": 90,
                "units": "0.5 FTE",
              },
            ],
            "statePersonnelTotal": 7006,
            "stateShare": 7007,
            "totalCost": 7008,
          },
        }
      }
    />
  </tbody>
</table>
`;
